src/components/vm/disks/diskAdd.jsx

```diff
@@ -583,6 +584,23 @@ export class AddDiskModalBody extends React.Component {
             storagePool = storagePools.find(pool => pool.name === this.state.storagePoolName);
             volume = storagePool.volumes.find(vol => vol.name === this.state.existingVolumeName);
             isVolumeUsed = getStorageVolumesUsage(vms, storagePool);
+        } else if (this.state.mode === CUSTOM_PATH) {
+            // fallback to raw.
+            format = "raw";
+            await cockpit.spawn(["qemu-img", "info", "--output", "json", this.state.file], { err: "message" })
+                    .then(output => {
+                        try {
+                            const obj = JSON.parse(output);
+                            if (obj.format) {
+                                format = obj.format;
+                            }
```

Comment:
There should also be an else here with a `console.error` -- we really don't expect this to not return a format, do we? Or can this happen with some unknown file types? (doesn't even an entirely blank file default to "raw"?)

Reply:
Ack, so let's log an error then, and default to `raw` ourselves in that case.